web/impact/impact/v1/views/office_hours_calendar_view.py

```diff
@@ -0,0 +1,116 @@
+from datetime import (
+    datetime,
+    timedelta,
+)
+
+from pytz import utc
+
+from rest_framework.response import Response
+
+from django.contrib.auth import get_user_model
+from django.db.models import (
+    Count,
+    F,
+)
+
+from . import ImpactView
+from accelerator.models import MentorProgramOfficeHour
+User = get_user_model()
+
+ISO_8601_DATE_FORMAT = "%Y-%m-%d"
+ONE_DAY = timedelta(1)
+ONE_WEEK = timedelta(8)
+
+
+class OfficeHoursCalendarView(ImpactView):
+    permission_classes = []
+    view_name = "office_hours_calendar_view"
+
+    SUCCESS_HEADER = "Placeholder text for success header"
+    FAILURE_HEADER = "Placeholder text for failure header"
+    BAD_DATE_SPEC = "We were unable to parse the date specifier"
+    NO_SUCH_USER = "We were not able to locate that user"
+
+    def fail(self, detail):
+        self.response_elements['success'] = False
+        self.response_elements['header'] = self.FAILURE_HEADER
+        self.response_elements['detail'] = detail
+        self.response_elements['calendar_data'] = None
+
+    def succeed(self):
+        self.response_elements['success'] = True
+        self.response_elements['header'] = self.SUCCESS_HEADER
+
+    def _get_target_user(self, request):
+        user_id = request.query_params.get("user_id", None)
+        if user_id is None:
+            self.target_user = request.user
+        else:
+            try:
+                self.target_user = User.objects.get(pk=user_id)
+            except User.DoesNotExist:
+                self.fail(self.NO_SUCH_USER)
+                return False
+        return True
+
+    def _get_start_date(self, request):
+        date_spec = request.query_params.get("date_spec", None)
+
+        try:
+            self.start_date = start_date(date_spec)
+        except ValueError:
+            self.fail(self.BAD_DATE_SPEC)
+            return False
+        return True
+
+    def _get_office_hours_data(self):
+        end_date = self.start_date + ONE_WEEK + 2 * ONE_DAY
+        office_hours = MentorProgramOfficeHour.objects.filter(
+             mentor=self.target_user,
+             start_date_time__range=[self.start_date, end_date]).order_by(
+                 'start_date_time').annotate(finalist_count=Count("finalist"))
+
+        self.response_elements['calendar_data'] = office_hours.values(
+            "id",
+            "start_date_time",
+            "end_date_time",
+            "mentor__last_name",
```

Comment:
Okay, working on this. 
A few points:
 
Re "title" and "company"
- Finalists do not have titles on the platform. Expert profiles have a "title" field, but no such field exists for finalists. 
- For finalists, "company" will be the startup selected by the finalist when they reserve the OH (if any)
- For experts, I can return "title" and "company" as separate fields for maximum flexibility on the front end

Reply:
Re: "meeting link", I'm not sure what this should be. No such field exists on the MentorProgramOfficeHour model.

Reply:
re: program_family, office hours created in the new epic are no longer associated with Programs, so there is no program_family to return

Reply:
re "location", I assume you mean location.name, ie "MassChallenge Boston" 
Can return full location details if you want (`location__name`, `location__street_address`, `location__city`, `location__state`, `location__postcode`)
Not sure what you need here because the mockup I can see has no location information showing.

Reply:
Is there any reason you want the full name concatenated? I prefer to return atomic data if possible, since that allows you more flexibility in formatting and requires less data munging on the server side.

Reply:
Including the timezone in the data bundle, but bear in mind that the current implementation calls for all date/times to be displayed in the TZ selected by the user (AC-7762, estimated today) so you will not need to know the timezone for any particular OH session. 

(in future we _may_ decide to display local time alongside selected time, but that's not on the docket for the MVP delivery) AFAIK

Reply:
> Okay, working on this.
> A few points:
> 
> Re "title" and "company"
> 
>     * Finalists do not have titles on the platform. Expert profiles have a "title" field, but no such field exists for finalists.
> 
>     * For finalists, "company" will be the startup selected by the finalist when they reserve the OH (if any)
> 
>     * For experts, I can return "title" and "company" as separate fields for maximum flexibility on the front end

True, but they do have a title via their startup team membership. Could that be used in this case?

Reply:
Sure, I can deliver that if you want

Reply:
> re: program_family, office hours created in the new epic are no longer associated with Programs, so there is no program_family to return

@hfreund

Reply:
Missing mentor primary industry

Reply:
> re "location", I assume you mean location.name, ie "MassChallenge Boston"
> Not sure what you need here because the mockup I can see has no location information showing.
I think returning `location.name` is acceptable.

Reply:
> re: program_family, office hours created in the new epic are no longer associated with Programs, so there is no program_family to return

While `program_family` is not associated with an office hour session, it does need to be displayed on a reservation. This is necessary to inform multi-program mentors in which program the startup who reserved their office hour is a finalist.

So in this case we're looking for the `program_family` of the startup or finalist who reserved the office hour.

Reply:
> Re: "meeting link", I'm not sure what this should be. No such field exists on the MentorProgramOfficeHour model.

`meeting_link` is a new field. If it has not yet been added through other work to the MentorPrograOfficeHour model, it probably ought to be. Are there other models where it should be added? Is this separate work or should it be done on this ticket?

Reply:
@hfreund record of our conversation:

- meeting_link will be added to the MentorProgramOfficeHour model in a future ticket, and that ticket will add meeting_link to the API output
- for now we will not include program_family in the office_hour output because it's complicated and easier to do later
- we'll be adding a list of location_choices to the output so the front-end will know what locations mentors can choose from